src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +113,68 @@ public ArmSubsystem() {
+        for (var config : new PID[] {ELBOW_DEFAULT, ELBOW_STEADY}) {
             m_elbow.getPIDController().setP(config.P, config.slot);
             m_elbow.getPIDController().setI(config.I, config.slot);
             m_elbow.getPIDController().setD(config.D, config.slot);
             m_elbow.getPIDController().setFF(config.FF, config.slot);
             m_elbow.getPIDController().setDFilter(config.DF, config.slot);
         }
 
+        m_elbow.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_elbow.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kReverse, true);
+
         // Set the soft limits
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
+    }
 
+    public void setState(State desiredState) {
+        this.m_desiredState = desiredState;
     }
 
-    public void setState(State state) {
-        // Don't do anything already in correct state
-        if (state == m_state_actual)
-            return;
-
-        switch (m_state_actual) {
-            case Pickup:
-                if (state == State.Rest)
-                    m_state_desired = state;
-                break;
-            case Rest:
-                m_state_desired = state;
-                break;
-            case PlacingA:
-            case PlacingB:
-            case PlacingC:
-            case PlacingD:
-            case PlacingE:
-                m_state_desired = state;
-                break;
-            case Transit:
-                break;
-        }
+    public boolean atState() {
+        boolean shoulder_at_state = goalCalculator(m_shoulder.getEncoder().getPosition(), ArmConstants.SHOULDER_GEAR_RATIO(m_desiredState.shoulder));
+        boolean elbow_at_state = goalCalculator(m_elbow.getEncoder().getPosition(), ArmConstants.ELBOW_GEAR_RATIO(m_desiredState.elbow));
+        return shoulder_at_state && elbow_at_state;
+    }
+
+    private boolean goalCalculator(double current, double desired) {
+        if (current == desired) return true;
+        return Math.abs(current - desired) <= 10;
     }
 
     public ArmSubsystem.State deserializeStateFromString(String stateName) {
         System.out.println("Deserializing state for " + stateName);
 
         switch (stateName) {
-            case "Rest": return State.Rest;
-            case "Pickup": return State.Pickup;
-            case "PlacingA": return State.PlacingA;
-            case "PlacingB": return State.PlacingB;
-            case "PlacingC": return State.PlacingC;
-            case "PlacingD": return State.PlacingD;
-            case "PlacingE": return State.PlacingE;
+            case "Rest":

```

Comment:
Revert this white space change, it serves no purpose.

Reply:
My formatter did this, but I'll try to revert it.

Reply:
Don't worry about it.